imagery-cesium/plugins/entities-visualizer/base-entities-visualizer.ts

```diff
@@ -25,45 +30,213 @@ export abstract class BaseEntitiesVisualizer extends BaseImageryVisualizer {
 
 	addOrUpdateEntities(logicalEntities: IVisualizerEntity[]): Observable<boolean> {
 		logicalEntities.forEach((entity: IVisualizerEntity) => {
-			if (entity.icon) {
-				const id = this.getIds(entity);
-				const billboardEntity = {
-					show: true,
-					position: Cesium.Cartesian3.fromDegrees(
-						entity.featureJson.geometry.coordinates[0],
-						entity.featureJson.geometry.coordinates[1],
-						entity.featureJson.geometry.coordinates[2] ? entity.featureJson.geometry.coordinates[2] : 0
-					),
-					pixelOffset: Cesium.Cartesian2.ZERO,
-					eyeOffset: Cesium.Cartesian3.ZERO,
-					heightReference: Cesium.HeightReference.NONE,
-					horizontalOrigin: Cesium.HorizontalOrigin.CENTER,
-					verticalOrigin: Cesium.VerticalOrigin.CENTER,
-					scale: 1.0,
-					image: entity.icon,
-					imageSubRegion: undefined,
-					color: Cesium.Color.WHITE,
-					id: id,
-					rotation: 0.0,
-					alignedAxis: Cesium.Cartesian3.ZERO,
-					width: undefined,
-					height: undefined,
-					scaleByDistance: undefined,
-					translucencyByDistance: undefined,
-					pixelOffsetScaleByDistance: undefined,
-					sizeInMeters: false,
-					distanceDisplayCondition: undefined
-				};
-				const feature = this.billboardCollection.add(billboardEntity);
-				this.idToEntity.set(id, <any>{
-					originalEntity: entity,
-					feature: feature
-				});
+			const featureJson: Feature<any> = entity.featureJson;
+
+			switch (featureJson.geometry.type) {
+				case 'Point': {
+					this.getOrCreateDataSource("Point").then(ds => {
+						if (entity.icon) {
+							ds.entities.add(this.getBillbord(entity.id, featureJson.geometry, entity.icon));
+						} else {
+							ds.entities.add(this.getPoint(entity.id, featureJson.geometry, entity.style));
+						}
+					});
+					break;
+				}
+				case 'LineString': {
+					this.getOrCreateDataSource("LineString").then(ds => {
+						ds.entities.add(this.getPolyline(entity.id, featureJson.geometry, entity.style));
+					});
+					break;
+				}
+				case 'Polygon': {
+					this.getOrCreateDataSource("Polygon").then(ds => {
+						ds.entities.add(this.getPolygon(entity.id, featureJson.geometry, entity.style));
+					});
+					break;
+				}
 			}
```

Comment:
add warning for all unsupported geometry type's
Add support for at least the following: 
multi polygon
multi Line
multi point